core/base/src/constants/finality.ts

```diff
@@ -0,0 +1,54 @@
+import { Network } from "./networks";
+import { ChainName } from "./chains";
+import { toMapping, toMappingFunc } from "../utils";
+
+const finalityThresholds = [
+  [
+    "Mainnet",
+    [
+      ["Ethereum", 64],
+      ["Solana", 32],
+      ["Polygon", 512],
+      ["Bsc", 15],
+      ["Avalanche", 1],
+      ["Fantom", 1],
+      ["Celo", 1],
+      ["Moonbeam", 1],
+      ["Sui", 0],
+      ["Aptos", 0],
+      ["Sei", 0],
+    ],
+  ],
+  [
+    "Testnet",
+    [
+      ["Ethereum", 64],
+      ["Solana", 32],
+      ["Polygon", 64],
+      ["Bsc", 15],
+      ["Avalanche", 1],
+      ["Fantom", 1],
+      ["Celo", 1],
+      ["Moonbeam", 1],
+      ["Sui", 0],
+      ["Aptos", 0],
+      ["Sei", 0],
+    ],
+  ],
+] as const satisfies readonly (readonly [
+  Network,
+  readonly (readonly [ChainName, number])[]
+])[];
+
+const mapping = {
+  Mainnet: toMapping(finalityThresholds[0][1]),
+  Testnet: toMapping(finalityThresholds[1][1]),
+  // TODO: reusing testnet finality, ok?
+  Devnet: toMapping(finalityThresholds[1][1]),
```

Comment:
I've been a little bit confused by Testnet and Devnet tbh... I'm wondering if we need a Devnet mapping or if we just have Mainnet and non-Mainnet? As far as I know, we usually pick one or the other, but we don't have any deployed to both.  And it's not super clear which chains are deployed on Testnet vs Devnet.  I'm not really recommending anything, just airing my grievances